From: Joachim Nilsson <troglobit@gmail.com>
Date: Mon, 27 Oct 2014 01:32:54 +0100
Subject: [PATCH] Fix issue #10: Loop forever on command if daemon not running.
Origin: https://github.com/troglobit/smcroute/commit/d6280e64b27d5a4bd7f37dac36b455f4ae5f9ab3
Bug: https://github.com/troglobit/smcroute/issues/10

Signed-off-by: Joachim Nilsson <troglobit@gmail.com>
---
 smcroute.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/smcroute.c b/smcroute.c
index 1aeb133..7bd3c72 100644
--- a/smcroute.c
+++ b/smcroute.c
@@ -573,10 +573,11 @@ int main(int argc, const char *argv[])
 		openlog(argv[0], LOG_PID, LOG_USER);
 
 		/* connect to daemon */
-		while (ipc_client_init()) {
+		while (ipc_client_init() && !result) {
 			switch (errno) {
 			case EACCES:
 				smclog(LOG_ERR, EACCES, "Need root privileges to connect to daemon");
+				result = 1;
 				break;
 
 			case ENOENT:
-- 
1.9.1

