Description: Fixed a segfault caused by dereferencing a NULL pointer
Author: Micha Lenk <micha@debian.org>
Bug-Debian: http://bugs.debian.org/707793
Bug-Ubuntu: https://launchpad.net/bugs/1043688
Forwarded: not-needed, fixed upstream by more intrusive code changes
Last-Update: 2013-05-14

--- smcroute-0.95.orig/src/ifvc.c
+++ smcroute-0.95/src/ifvc.c
@@ -50,12 +50,13 @@ void buildIfVc()
   }
 
   for (ifa = ifaddr; ifa != NULL; ifa = ifa->ifa_next) {
+    /* Skip interface without internet address */
+    if (ifa->ifa_addr == NULL) continue;
+
     int family = ifa->ifa_addr->sa_family;
 
     /* Skip non-IPv4 and non-IPv6 interfaces */
     if ((family != AF_INET) && (family != AF_INET6)) continue;
-    /* Skip interface without internet address */
-    if (ifa->ifa_addr == NULL) continue;
 
     /* Copy data from interface iterator 'ifa' */
     strncpy(IfDescEp->Name, ifa->ifa_name, sizeof(IfDescEp->Name));
